Compiler bug when returned arg is a pointer using a builtin method

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Ubuntu 22.04 LTS
Processor: 12 cpus, 64bit, little endian, Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
CC version: cc (Ubuntu 11.2.0-19ubuntu1) 11.2.0

V full version: V 0.3.0 eb10cc9.426421b
Git version: git version 2.34.1
Git vroot status: 0.3-56-g426421be
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 827f7452
```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg /var/www/vproject/test/0sample.5.bug.v`
```v
module main

fn main() {
	println(yolen())
}

fn yolen() &int {
	return &'yo'.len
}

```

**What did you expect to see?**

The size of string len


**What did you see instead?**
```
==================
/tmp/v_1000/0sample.5.bug.10511335850393822228.tmp.c:12078: error: lvalue expected
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

Simplified version:

```v
>>> &"".len
==================
/tmp/v_1000/.noprefix.01G7CY15YX7D5FMN43BWMGW6NT.vrepl_temp.14939082539941444648.tmp.c:21416: error: lvalue expected
...
==================
(Use `v -cg` to print the entire error message)

builder error:
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

The reason this error occurs is because the `transformer` converts `str_lit.len` to an integer literal whose value is the `.len` of the string literal, so the `cgen` outputs `&0`, which is invalid in C.

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -3906,6 +3906,9 @@ fn (mut c Checker) prefix_expr(mut node ast.PrefixExpr) ast.Type {
 				c.error('unexpected `&`, expecting expression', node.right.pos)
 			}
 		} else if mut node.right is ast.SelectorExpr {
+			if node.right.expr.is_literal() {
+				c.error('cannot take the address of a literal value', node.pos.extend(node.right.pos))
+			}
 			right_sym := c.table.sym(right_type)
 			expr_sym := c.table.sym(node.right.expr_type)
 			if expr_sym.kind == .struct_ && (expr_sym.info as ast.Struct).is_minify

--- a/vlib/v/checker/tests/prefix_addr_err.out
+++ b/vlib/v/checker/tests/prefix_addr_err.out
@@ -1,0 +1,10 @@
+vlib/v/checker/tests/prefix_addr_err.vv:2:2: warning: unused variable: `b`
+    1 | fn main() {
+    2 |     b := &'foo'.len
+      |     ^
+    3 | }
+vlib/v/checker/tests/prefix_addr_err.vv:2:7: error: cannot take the address of a literal value
+    1 | fn main() {
+    2 |     b := &'foo'.len
+      |          ~~~~~~~~~~
+    3 | }

--- a/vlib/v/checker/tests/prefix_addr_err.vv
+++ b/vlib/v/checker/tests/prefix_addr_err.vv
@@ -1,0 +1,3 @@
+fn main() {
+	b := &'foo'.len
+}

